client/html/src/Client/Html/Base.php

```diff
@@ -707,8 +707,13 @@ protected function translatePluginErrorCodes( array $codes )
 		{
 			foreach( $list as $object => $errcode )
 			{
-				$key = $scope . ( $scope !== 'product' ? '.' . $object : '' ) . '.' . $errcode;
-				$errors[] = $i18n->dt( 'mshop/code', $key );
+				if( $scope === 'product' || $scope === 'coupon' ) {
```

Comment:
This should do the trick:
```
$key = $scope . ( !in_array( $scope, ['coupon', 'product'] ) ? '.' . $object : '' ) . '.' . $errcode;
$errors[] = sprintf( $i18n->dt( 'mshop/code', $key ), $object );
```
You want to build "coupon.gone" and replace the code if theres a `%1$s` in the string.